### ISSUE TITLE
Running the Code

### ISSUE THREAD

**Issue by ghiu-75** [jpcurbelo/human-body-reshape-DL-paper#6]
Q1) If I just want to get anthropometric measurements from 2d images of human (female/male);
I will -> Go to src/extractor and run: python extractor_model_training.py
And I will get the anthropometric measurements.
Is that right?
Q2) If I have some anthropometric measurements (height/ weight/Bust (Chest)/ waist/hips) and images of(female/male) and add use these for training purposes
and then I want to get anthropometric measurements from 2d images of human (female/male); What should I do?

**Comment by jpcurbelo** [comment]
Hi @ghiu-75. I'm copy/pasting my email here just for the records:

Thanks for your interest in our code. I'll try to answer your questions here but feel free to reach me back in case I left anything unclear. 

Q1) Yes, once you set up the environment and download the needed data files, extractor_model_training.py is the entry point to the application. The call

```
extracted_measurements = measurements_from_sil(
   model,
   input_info, 
   sil_images,
   df_total,
   stature_glob,
)
```
extracts 8 body measurements from the images, height, and weight - weight can be not informed though. 

Then a set of 21 measurements is completed by imputation and the 3D avatar is generated. In addition to the `wavefront.obj` file for the avatar, an output table (`output_data_avatar_male_fromImg.csv` attached) is generated containing the measurements values in the different states: Basic-Input (after extraction from images), Predicted-Input (after imputation), and 3D Avatar-Output (by measuring the 3D avatar).

Q2) The NN to extract body measurements was trained with 2 full-body images, height, and weight as input data, and chest_girth, waist_girth, hips_buttock_girth, shoulder_girth, thigh_girth sleeveoutseam_length, crotchheight_length, and waistback_length as target values. Although, the code could be adapted to use whatever set of measurements you might have depending on the database you want/need to use. 

I hope this helps. I recommend starting by running the code step by step as outlined in the Readme file. This can help you get familiar with the code and get a feel for how everything works before making any modifications.
![image](https://github.com/user-attachments/assets/3eb432bd-7113-4d99-9497-39d8fb6e4f02)

**Comment by jpcurbelo** [comment]
About your following up comments:

_1) For weight estimation, what should your suggestion be?_

One does not need the weight to be an input data to run the Extractor (only images and height are mandatory). Although, the weight can be useful for the imputation of the "missing" 10 measurements after the extraction of the main measurements out of the images (+ height). That is why informing the weight is recommended but not mandatory.

_2) can I train the program with images of (f/m) just with their height and weight info?_

Although the NN model (Extractor) is intended to use Images+Height (the weight is to be used during the Imputation), it could be adapted to only use Images. This is an experiment that we run in the paper, and we noticed that the accuracy (according to the metric we used) was reduced from 97% to 91%.
